app/assets/scripts/views/tasks.js

```diff
@@ -0,0 +1,334 @@
+'use strict';
+
+import React from 'react';
+import { connect } from 'react-redux';
+import mapboxgl from 'mapbox-gl';
+import config from '../config';
+import getExtent from 'turf-extent';
+import { t } from '../utils/i18n';
+
+import {
+  fetchWayTasks
+} from '../actions/action-creators';
+
+const source = 'collisions';
+const roadHoverId = 'road-hover';
+const collisionHoverId = 'collision-hover';
+const layers = [{
+  id: 'road',
+  type: 'line',
+  paint: {
+    'line-width': 6,
+    'line-opacity': 0.8,
+    'line-color': '#FCF009'
+  },
+  layout: { 'line-cap': 'round' },
+  filter: ['has', '_main']
+}, {
+  id: 'collision',
+  type: 'line',
+  paint: {
+    'line-width': 4,
+    'line-opacity': 0.2
+  },
+  layout: { 'line-cap': 'round' },
+  filter: ['!has', '_main']
+}, {
+  id: roadHoverId,
+  type: 'line',
+  paint: {
+    'line-width': 8,
+    'line-opacity': 0.9,
+    'line-color': '#FCF009'
+  },
+  layout: { 'line-cap': 'round' },
+  filter: ['all', ['has', 'main'], ['==', '_id', '']]
+}, {
+  id: collisionHoverId,
+  type: 'line',
+  paint: {
+    'line-width': 6,
+    'line-opacity': 0.9
+  },
+  layout: { 'line-cap': 'round' },
+  filter: ['all', ['!has', 'main'], ['==', '_id', '']]
+}].map(layer => Object.assign({source}, layer));
+
+const layerIds = layers.map(layer => layer.id);
+
+var Tasks = React.createClass({
+  getInitialState: function () {
+    return {
+      currentTaskId: null,
+      currentTask: null,
```

Comment:
It's a bit redundant to store both the task and the id. Can't we just store the task?